staging/src/k8s.io/apiserver/pkg/endpoints/audit_test.go

```diff
@@ -0,0 +1,326 @@
+			},
+			selfLinker,
+			400,
+			1,
+			[]eventCheck{
+				requestBodyIs(0, string(simpleCPrimeJSON)),
+				responseBodyIs(0, ""),
+			},
+		},
+		{
+			"patch",
+			func(server string) (*http.Request, error) {
+				req, _ := http.NewRequest("PATCH", server+"/"+prefix+"/"+testGroupVersion.Group+"/"+testGroupVersion.Version+"/namespaces/other/simple/c", bytes.NewReader([]byte(`{"labels":{"foo":"bar"}}`)))
+				req.Header.Set("Content-Type", "application/merge-patch+json; charset=UTF-8")
+				return req, nil
+			},
+			&setTestSelfLinker{
+				t:           t,
+				expectedSet: "/" + prefix + "/" + testGroupVersion.Group + "/" + testGroupVersion.Version + "/namespaces/other/simple/c",
+				name:        "c",
+				namespace:   "other",
+			},
+			200,
+			1,
+			[]eventCheck{
+				requestBodyIs(0, `{"labels":{"foo":"bar"}}`),
+				responseBodyMatches(0, `"name":"c".*"labels":{"foo":"bar"}`),
+			},
+		},
+		{
+			"watch",
+			func(server string) (*http.Request, error) {
+				return http.NewRequest("GET", server+"/"+prefix+"/"+testGroupVersion.Group+"/"+testGroupVersion.Version+"/namespaces/other/simple?watch=true", nil)
+			},
+			&setTestSelfLinker{
+				t:           t,
+				expectedSet: "/" + prefix + "/" + testGroupVersion.Group + "/" + testGroupVersion.Version + "/namespaces/other/simple",
+				namespace:   "other",
+			},
+			200,
+			2,
+			[]eventCheck{
+				requestBodyMatches(0, ""),
+				responseBodyMatches(0, ""),
+			},
+		},
+	} {
+		sink := &fakeAuditSink{}
+		handler := handleInternal(map[string]rest.Storage{
+			"simple": &SimpleRESTStorage{
+				list: []genericapitesting.Simple{
+					{
+						ObjectMeta: metav1.ObjectMeta{Name: "a", Namespace: "other"},
+						Other:      "foo",
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{Name: "b", Namespace: "other"},
+						Other:      "foo",
+					},
+				},
+				item: genericapitesting.Simple{
+					ObjectMeta: metav1.ObjectMeta{Name: "c", Namespace: "other", UID: "uid"},
+					Other:      "foo",
+				},
+			},
+		}, admissionControl, selfLinker, sink)
+
+		server := httptest.NewServer(handler)
+		defer server.Close()
+		client := http.Client{Timeout: 2 * time.Second}
+
+		req, err := test.req(server.URL)
+		if err != nil {
+			t.Errorf("[%s] error creating the request: %v", test.desc, err)
+		}
+
+		response, err := client.Do(req)
+		if err != nil {
+			t.Errorf("[%s] error: %v", test.desc, err)
+		}
+
+		if response.StatusCode != test.code {
+			t.Errorf("[%s] expected http code %d, got %#v", test.desc, test.code, response)
+		}
+
+		// close body because the handler might block in Flush, unable to send the remaining event.
+		response.Body.Close()
+
+		// wait for events to arrive, at least the given number in the test
+		events := []*auditinternal.Event{}
+		err = wait.Poll(50*time.Millisecond, 1*time.Second, wait.ConditionFunc(func() (done bool, err error) {
```

Comment:
actually we have this 1 second for a number of unit tests, just fixed a couple some days ago (they had <<1s and flaked). But with `wait.ForeverTestTimeout` we are on the safe side 👍